### DISCUSSION TITLE
Switch from builtin modulebuilder to puppet-modulebuilder

### DISCUSSION THREAD

**Discussion by bastelfreak** [puppetlabs/pdk#1360] [Feature Request] [answered]
We had a few conversations in the past about pdk and how we can increase the adoption rate in the community. A long time ago people were able to run `puppet module build` to build a puppet module. That functionality is now in https://github.com/puppetlabs/puppet-modulebuilder. pdk doesn't use that gem but reinvents the wheel / shares some code: https://github.com/puppetlabs/pdk/blob/main/lib/pdk/module/build.rb

I think it makes sense for pdk to also use puppet-modulebuilder because that will ensure that pdk behaves the same when building modules. Also that removes some maintenance burden from DevX. For example new features like https://github.com/puppetlabs/puppet-modulebuilder/pull/79 don't need to be implemented into pdk again when it uses puppet-modulebuilder.

**Comment by LukasAud** [comment]
We think it's a good idea and will look into it to see if it's viable for us.

**Comment by david22swan** [comment, marked as answer]
As confirmation, this has been added to our backlog and scheduled to be done.
